stdlib/public/core/Codable.swift

```diff
@@ -2894,13 +2894,28 @@ public extension RawRepresentable where RawValue == String, Self : Decodable {
 //===----------------------------------------------------------------------===//
 
 // FIXME: Uncomment when conditional conformance is available.
-extension Array : Codable /* where Element : Codable */ {
-    public init(from decoder: Decoder) throws {
-        guard Element.self is Decodable.Type else {
-            preconditionFailure("Array<\(Element.self)> does not conform to Decodable because \(Element.self) does not conform to Decodable.")
+extension Array : Encodable /* where Element : Encodable */ {
```

Comment:
Basically today you would be able to define a generic function that takes an array and encodes it. Once conditional conformance comes in, you would need to add a `where` clause to that function in order for the passed-in array to be encodable to pass compile-time checks, even though you were only ever using it with arrays that passed the checks at run-time.